src/RobotHub.cpp

```diff
@@ -151,6 +155,72 @@ void RobotHub::onSettings(const proto::Setting &settings) {
     this->mode = newMode;
 }
 
+void RobotHub::onSimulationConfiguration(const proto::SimulationConfiguration &configuration) {
```

Comment:
Why is everything copied over? Can't the packet simply be forwarded?

Reply:
The proto::SimulationConfiguration is not a proto file that can be sent to the Simulator. The proto packets that the simulator receives are a bit more complex unfortunately, so I chose to make a simpler proto packet that we can use in our repo, which will be converted to the complex one later on.
However, it is true we could forward this proto file at least a little untill it is directly converted to the complex proto packet instead of first converting it to a c++ object, but that would mean the simulator manager would also be dependent on our specific proto, which is exactly the thing I was trying to avoid. Currently, proto is deeply entangled into rtt_world and I assume also in rtt_AI. If you then would want to change a proto file, you would really have to dive deep into all repositories and change variable names everywhere, instead of just at the beginning where the proto is converted to a datatype that repo understands.
I understand that in this specific situation converting it twice might be a bit redundant, as the travel distance of this message is about 2 functions, but there is no need for us to change it, as efficiency will not be a bottleneck.